{{cookiecutter.project_name}}/notebooks/ModelValidation.py

```diff
@@ -0,0 +1,224 @@
+# MAGIC %autoreload 2
+
+# COMMAND ----------
+
+# MAGIC %pip install -r ../requirements.txt
+
+# COMMAND ----------
+
+dbutils.widgets.dropdown("env", "prod", ["staging", "prod"], "Environment Name")
+dbutils.widgets.dropdown("run_mode", "Disabled", ["Disabled", "Dry Run", "Enabled"], "Run Mode")
+
+# COMMAND ----------
+
+import mlflow
+from mlflow.models import MetricThreshold, make_metric
+import pandas as pd
+import os
+import numpy as np
+import tempfile
+import json
+import traceback
+from mlflow.recipes.utils import (
+    get_recipe_config,
+    get_recipe_name,
+    get_recipe_root_path,
+)
+from mlflow.tracking.client import MlflowClient
+client = MlflowClient()
+
+env = dbutils.widgets.get("env")
+_run_mode = dbutils.widgets.get("run_mode")
+if _run_mode.lower() == "disabled":
+    dbutils.notebook.exit(0)
+dry_run = _run_mode.lower() == "dry run"
+
+def get_model_type_from_recipe():
+    recipe_config = get_recipe_config("../", f"databricks-{env}")
+    problem_type = recipe_config.get("recipe").split("/")[0]
+    if problem_type.lower() == "regression":
+        return "regressor"
+    elif problem_type.lower() == "classification":
+        return "classifier"
+    else:
+        raise Exception(f"Unsupported recipe {recipe_config}")
+
+def get_targets_from_recipe():
+    recipe_config = get_recipe_config("../", f"databricks-{env}")
+    return recipe_config.get("target_col")
+
+# set model evaluation parameters that can be inferred from the job
+model_uri = dbutils.jobs.taskValues.get("Train", "model_uri", debugValue="")
+model_name = dbutils.jobs.taskValues.get("Train", "model_name", debugValue="")
+model_version = dbutils.jobs.taskValues.get("Train", "model_version", debugValue="")
+
+baseline_model_uri = "models:/" + model_name + "/Production"
+evaluators = "default"
+assert env != "None", "env notebook parameter must be specified"
+assert model_uri != "", "model_uri notebook parameter must be specified"
+assert model_name != "", "model_name notebook parameter must be specified"
+assert model_version != "", "model_version notebook parameter must be specified"
+
+# set experiment
+with open("../databricks-config/output/prod.json", "r") as f:
+    experiment_name = json.load(f)["{{cookiecutter.project_name}}_experiment_name"]["value"]
+mlflow.set_experiment(experiment_name)
+
+# COMMAND ----------
+
+##################################################################################
+# TODO : Please fill in this cell with proper values for enable_baseline_comparison,
+#        data, targets, model_type, custom_metrics, validation_thresholds
+#        evaluator_config
+##################################################################################
+
+# Whether to load the current registered "Production" stage model as baseline. A version with "Production" stage must 
+# exist for the model.
+# Baseline model is a requirement for relative change and absolute change validation rules.
+# TODO(required) : enable_baseline_comparison
```

Comment:
Seems like it's already enabled?

Reply:
Good catch. Default should be false.

Reply:
Why is it a `required` TODO instead of `optional`?

Reply:
It is required to set the value for this variable. The value can be true or false.
It is required because we are using the variable value later.

Reply:
Ok I think it can be optional.